src/Columns/ColumnArray.cpp

```diff
@@ -134,10 +134,11 @@ Field ColumnArray::operator[](size_t n) const
         throw Exception(ErrorCodes::TOO_LARGE_ARRAY_SIZE, "Array of size {} is too large to be manipulated as single field, maximum size {}",
             size, max_array_size_as_field);
 
-    Array res(size);
+    Array res;
+    res.reserve(size);
 
     for (size_t i = 0; i < size; ++i)
-        res[i] = getData()[offset + i];
+        res.push_back(getData()[offset + i]);
```

Comment:
Previous code was ok. Maybe change like this `getData().get(offset + i, res[i])`.